flask_app/tasks/replications.py

```diff
@@ -302,7 +302,7 @@ def _get_properties_definitions(self):
 
     def enrich_result(self, result):
         next_keepalive = result.get('next_keepalive')
-        result['is_abandoned'] = not (result.get('end_time') or next_keepalive is None or next_keepalive > get_current_time())
+        result['is_abandoned'] = (next_keepalive is not None and next_keepalive < get_current_time()) or (result.get('end_time') is None)
```

Comment:
So if we have a next_keepalive, and it's in 30 seconds... end_time will be None so we will deduce this is abandoned?!

Reply:
updated.